### PR TITLE
Keep the players and stats. 📃

### PR DESCRIPTION
**Pull request by Ashthecool**
* Made some main menu changes.
* Made some new UI sprites. 
 
---

Future:
* Make a list of Players and keep them from level to level.

### CODE REVIEW

File: my_game.py [hunk 1]
```diff
@@ -385,6 +415,11 @@ def on_show_view(self):
         # to reset the viewport back to the start so we can see what we draw.
         arcade.set_viewport(0, self.window.width, 0, self.window.height)
 
+        self.player_amount = 1
+        self.player_sprite_list = arcade.SpriteList()
+
+        self.player_sprite_list = create_players(self.player_amount)
```
Comment (tobiasbp) [inline my_game.py:421]: Huh? I thought the number of players could be selected. Here, it's hard coded to 1??

File: my_game.py [hunk 2]
```diff
@@ -400,22 +435,35 @@ def on_show_view(self):
...
-            x=150,
+        self.one_player = arcade.gui.UITextureButton(
+            x=50,
             y=125,
             width=100,
             height=100,
-            texture=arcade.load_texture("images/GUI/start_button_unhovered.png"),
-            texture_hovered=arcade.load_texture("images/GUI/start_button_hovered.png"),
-            scale=button_scaling,
+            texture=arcade.load_texture("images/GUI/one_player.png"),
```
Comment (tobiasbp) [inline my_game.py:443]: Could we rename the button files to something like _players_one_chosen.png_ & _players_one.png_.

File: my_game.py [hunk 3]
```diff
@@ -446,12 +505,33 @@ def on_key_press(self, key: int, modifiers: int):
...
+
+        self.player_amount = 2
+        self.two_player.on_click = self.start_game
+
     def start_game(self, event=None):
         """
         Starts the game.
         """
+
+        # FIXME: Create the players in a for loop. Add the to the list!
```
Comment (tobiasbp) [inline my_game.py:529]: The players should be created here! How else do you know how many to create?

File: my_game.py [hunk 4]
```diff
@@ -446,12 +505,33 @@ def on_key_press(self, key: int, modifiers: int):
...
+
+        # Prevent the sound from playing after the game starts
         self.opening_sound.stop(self.opening_sound_player)
-        game_view = GameView(0)
+        print("The amount of Players are:", self.player_amount)
+        game_view = GameView(0, player_sprite_list=self.player_sprite_list)
```
Comment (tobiasbp) [inline my_game.py:534]: Use argument name for both or for none of them.

**Review by tobiasbp — CHANGES_REQUESTED** [review]
Need to be bale to select number of players. Otherwise good.

File: my_game.py (hunk 2 above)
Comment (tobiasbp) [reply my_game.py:443]: Do the same for other files used for buttons, so they all have the prefix _button__.

File: my_game.py [hunk 5]
```diff
@@ -74,16 +74,46 @@
...
+    return player_sprite_list
+
+
 class GameView(arcade.View):
     """
     The view with the game itself
     """
     
-    def __init__(self, level):
+    def __init__(self, level, player_sprite_list):
```
Comment (tobiasbp) [inline my_game.py:112]:
```suggestion
        '''
level: The level number to load
player_sprite_list: The Players to add to the level
'''
```

File: my_game.py [hunk 6]
```diff
@@ -74,16 +74,46 @@
...
         
         super(GameView, self).__init__()
```
Comment (tobiasbp) [inline my_game.py:113]: Add a _doc string: to the _init()_:
´´´
'''
level: The level number to load
player_sprite_list: The Players to add to the level
'''
´´´

File: my_game.py [hunk 7]
```diff
@@ -147,26 +181,18 @@ def on_show_view(self):
 
         self.player_score = 0
 
-        self.player_sprite_list = []
+        # self.player_sprite_list = []
```
Comment (tobiasbp) [inline my_game.py:184]: Remove comment

File: my_game.py [hunk 8]
```diff
@@ -147,26 +181,18 @@ def on_show_view(self):
...
 
         # replace all sprites on layer "players" with actual player objects
-        for i in range(NUM_OF_PLAYERS):
+
+        for i in range(len(self.player_sprite_list)):
             # Creates Player object
```
Comment (tobiasbp) [inline my_game.py:189]: Lying comment!

File: my_game.py [hunk 9]
```diff
@@ -147,26 +181,18 @@ def on_show_view(self):
...
-                scale=SCALING,
-                key_up=PLAYER_KEYS[i]["up"],
-                key_down=PLAYER_KEYS[i]["down"],
-                key_left=PLAYER_KEYS[i]["left"],
-                key_right=PLAYER_KEYS[i]["right"],
-                key_attack=PLAYER_KEYS[i]["attack"],
+
+            self.player_sprite_list[i].position = (
+                self.tilemap.sprite_lists["players"][0].center_x + random.randint(1,8) * TILE_SIZE * SCALING,
+                self.tilemap.sprite_lists["players"][0].center_y
```
Comment (tobiasbp) [inline my_game.py:193]: We should spawn at the dedicated tiles. We have a tile for each player, right?

**Review by tobiasbp — CHANGES_REQUESTED** [review]

File: my_game.py (hunk 9 above)
Comment (Ashthecool) [reply my_game.py:187]: Yes we do!

**Review by tobiasbp — APPROVED** [review]

**Pull request merged**